src/Module/Api/Mastodon/FollowRequests.php

```diff
@@ -78,41 +78,23 @@ public static function rawContent(array $parameters = [])
 
 		$baseUrl = DI::baseUrl();
 
-		if (isset($since_id) && isset($max_id)) {
-			$condition = ['`uid` = ? AND NOT `ignore` AND `id` > ? AND `id` < ?', self::$current_user_id, $since_id, $max_id];
-		} elseif (isset($since_id)) {
-			$condition = ['`uid` = ? AND NOT `ignore` AND `id` > ?', self::$current_user_id, $since_id];
-		} elseif (isset($max_id)) {
-			$condition = ['`uid` = ? AND NOT `ignore` AND `id` < ?', self::$current_user_id, $max_id];
-		} else {
-			$condition = ['`uid` = ? AND NOT `ignore`', self::$current_user_id];
-		}
-
-		$count = DBA::count('intro', $condition);
-
-		$intros = DBA::selectToArray(
-			'intro',
-			[],
-			$condition,
-			['order' => ['id' => 'DESC'], 'limit' => $limit]
+		$Introductions = DI::intro()->selectByBoundaries(
```

Comment:
I think I realize what are you doing with the `Collection` class, but it seems a little bit oversized.. You use it to can have abstract calls but nevertheless get an type-safe model inside the collection class you return, isn't it?

I would prefer overriding the method and return an array `Introduction[]`, so we don't need an extra class for it.

Reply:
Yes, type safety is key to me. Makes code fool-proof I believe.

Reply:
And you're probably right ..